src/modules/control_allocator/ControlAllocation/ControlAllocation.cpp

```diff
@@ -115,34 +117,27 @@ ControlAllocation::setActuatorSetpoint(
 	_actuator_sp = actuator_sp;
 
 	// Clip
-	_actuator_sp = clipActuatorSetpoint(_actuator_sp);
+	clipActuatorSetpoint(_actuator_sp);
 
 	// Compute achieved control
 	_control_allocated = _effectiveness * _actuator_sp;
 
 }
 
-matrix::Vector<float, ControlAllocation::NUM_ACTUATORS>
-ControlAllocation::clipActuatorSetpoint(const matrix::Vector<float, ControlAllocation::NUM_ACTUATORS> &actuator) const
+void
+ControlAllocation::clipActuatorSetpoint(matrix::Vector<float, ControlAllocation::NUM_ACTUATORS> &actuator) const
 {
-	matrix::Vector<float, ControlAllocation::NUM_ACTUATORS> actuator_clipped;
-
-	for (size_t i = 0; i < ControlAllocation::NUM_ACTUATORS; i++) {
+	for (int i = 0; i < _num_actuators; i++) {
```

Comment:
It depends on how `_num_actuators` is interpreted. I intended it as a maximum, and actuators can still be inactive within that (which might dynamically change, in which case I would not want to change the runtime behavior). I'm aware the name does not reflect that, I'm open to better suggestions.
In the bigger scheme I still want to look into the overall design in terms of parameters, and make a proposal. This might include a count parameter for each actuator type to make this more explicit.

Reply:
> and actuators can still be inactive within that

Hum ok my bad, I read it `num_actuators += 1;` instead of `num_actuators = i + 1;` in ActuatorEffectivenessMultirotor::computeEffectivenessMatrix().

I find it a bit misleading but I do not have a better solution.